### PR TITLE
[CircleCI] Extended existing try-catch block

### PR DESCRIPTION
**Pull request by PyvesB**
This pull request should resolve issue #1608.

### CODE REVIEW

**Comment by PyvesB** [comment]
I believe the CI failure is unrelated to this PR.

**Comment by RedSparr0w** [comment]
May be worth adding the [`checkErrorResponse()`](https://github.com/badges/shields/blob/master/server.js#L570-L573) helper function,
As it looks like `data` is undefined so i'm assuming this is happening when the request is returning a `404` or `!=200` status code possibly?

**Comment by paulmelnikow** [comment]
By the way, to re-run a test in CI, log in to Circle using your Github account. Look for the Workflow **on-commit** link and click it. From the dropdown, choose **Rerun failed jobs**.

**Comment by PyvesB** [comment]
@RedSparr0w I've added the `checkErrorResponse` helper function. Not entirely sure why `data` would be undefined. Status codes should now be handled by the helper, I've added a test where we get a 200 but no response data at all. 😉 

@paulmelnikow thanks for the tip!

File: services/circleci/circleci.tester.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ t.create('circle ci (valid, with branch)')
 
 t.create('circle ci (not found)')
   .get('/project/github/PyvesB/EmptyRepo.json')
-  .expectJSON({name: 'build', value: 'Project not found'});
+  .expectJSON({name: 'build', value: 'not found'});
```
Comment (paulmelnikow) [inline services/circleci/circleci.tester.js]: I like the explicit message. You can pass a custom "not found" message to `checkErrorResponse` to preserve the old behavior.

Comment (PyvesB) [reply services/circleci/circleci.tester.js]: Reverted. I've nevertheless switched to a lower case `P` to be more consistent with what we use elsewhere.

**Comment by RedSparr0w** [comment]
>Not entirely sure why data would be undefined.

Actually think i was partially wrong about the status code being the problem with it, Think it may have to do with `{json:true}` in the request, if the data that is returned is not JSON then it probably just returns `undefined`.
_(just a guess, haven't had a chance to test it out yet)_

**Pull request merged**